docs/GETTING_STARTED.md

```diff
@@ -1,288 +1,242 @@
+Many `soci` CLI commands need to be run as `sudo`, because the metadata is saved
+in directories that a non-root user often does not have access to.
 
-> Registries are evolving as generic artifact stores. To enable this goal, the ORAS project provides a way to push and pull OCI Artifacts to and from OCI Registries.
+Currently to get the binaries, we need to build the project from source after cloing the repo:
 
-[The longer term solution for this issue is using the OCI Reference
-Types](https://github.com/awslabs/soci-snapshotter/issues/42).  For now we are
-relying on the ORAS project for the referrers feature.
+```shell
+git clone https://github.com/awslabs/soci-snapshotter.git
+cd soci-snapshotter
+make
+```
 
-Given this we want to start our registry using an OCI image from the ORAS 
-project.  We will run it on port 5000 as an HTTP (**NOT** HTTPS) server without
-certificates.  To do this we will use Docker:
+This builds the project binaries into the `./out` directory. You can install them
+to a `PATH` directory (`/usr/local/bin`) with:
 
-```
-docker run -d -p 5000:5000 --restart=always --name registry ghcr.io/oci-playground/registry:v3.0.0-alpha.1
+```shell
+sudo make install
 ```
 
-Since this is just a locally run HTTP server we can interact with
-it using standard `curl` commands. To confirm this is running lets
-use the following commands:
+> This doc assumes SOCI binaries are installed into `PATH`. If not, please use
+> the full path of the binaries (e.g. `./out/soci`).
 
-```
-docker ps
-```
+## Push an image to your registry
 
-which should yield a running container at 0.0.0.0:5000 for the ORAS image. Now
-to check that we can interact with the server run the following:
-```
-curl http://localhost:5000/v2/_catalog
-```
-This should return with `{"repositories":[]}` given that we haven't put anything
-in there yet
+In this document we will use `rabbitmq` from DockerHub `docker.io/library/rabbitmq:latest`.
+We use [AWS ECR](https://docs.aws.amazon.com/AmazonECR/latest/userguide/getting-started-console.html)
+as the public registry for demonstration. Other OCI 1.0 compatible registries should
+also work.
 
-## Push an image to the registry using ctr
-First things first, let's choose our favorite image.  For the purpose of this
-document we will use the latest image of rabbitmq from DockerHub, aptly named
-`docker.io/library/rabbitmq:latest`.
+> You need to change `REGISTR` to your own registry.
+> The platform tag might be different depending on your machine.
 
-To do this with ctr use the following command:
-```
-ctr i pull docker.io/library/rabbitmq:latest
+```shell
+export AWS_ACCOUNT=000000000000
+export AWS_REGION=us-east-1
+export REGISTR=$AWS_ACCOUNT.dkr.ecr.$AWS_REGION.amazonaws.com
+sudo ctr i pull docker.io/library/rabbitmq:latest
+sudo ctr i tag docker.io/library/rabbitmq:latest /rabbitmq:latest
+sudo ctr i push --platform linux/amd64 $REGISTRY/rabbitmq:latest
```

Comment:
Since we are pushing/pulling from private repositories (ECR) we should also add the `--user AWS:$ECR_PASS` bit to our push and pull commands.

Reply:
We can also export ECR_PASS (in the case of ecr `export ECR_PASS=$(aws ecr get-login-password --region <REGION>`